### PR TITLE
Rename mentions of Redhawk to NaitveAOT.

### PR DESCRIPTION
**Pull request by prasanta10**

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/CMakeLists.txt [hunk 1]
```diff
@@ -233,7 +233,7 @@ add_definitions(-DFEATURE_CONSERVATIVE_GC)
 add_definitions(-DFEATURE_CUSTOM_IMPORTS)
 add_definitions(-DFEATURE_DYNAMIC_CODE)
 add_compile_definitions($<$<OR:$<CONFIG:Debug>,$<CONFIG:Checked>>:FEATURE_GC_STRESS>)
-add_definitions(-DFEATURE_REDHAWK)
+add_definitions(-DFEATURE_NativeAOT)
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/nativeaot/Runtime/CMakeLists.txt:236]:
```suggestion
add_definitions(-DFEATURE_NATIVEAOT)
```

File: src/coreclr/nativeaot/Runtime/EtwEvents.h [hunk 2]
```diff
@@ -70,7 +70,7 @@ __declspec(noinline) __inline bool __stdcall
 
 #define ETW_EVENT_ENABLED(Context, EventDescriptor) (Context.IsEnabled && RhEventTracingEnabled(&Context, &EventDescriptor))
 
-extern "C" __declspec(selectany) const GUID MICROSOFT_WINDOWS_REDHAWK_GC_PRIVATE_PROVIDER = {0x1095638c, 0x8748, 0x4c7a, {0xb3, 0x9e, 0xba, 0xea, 0x27, 0xb9, 0xc5, 0x89}};
+extern "C" __declspec(selectany) const GUID MICROSOFT_WINDOWS_NativeAOT_GC_PRIVATE_PROVIDER = {0x1095638c, 0x8748, 0x4c7a, {0xb3, 0x9e, 0xba, 0xea, 0x27, 0xb9, 0xc5, 0x89}};
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/nativeaot/Runtime/EtwEvents.h:73]:
```suggestion
extern "C" __declspec(selectany) const GUID MICROSOFT_WINDOWS_NATIVEAOT_GC_PRIVATE_PROVIDER = {0x1095638c, 0x8748, 0x4c7a, {0xb3, 0x9e, 0xba, 0xea, 0x27, 0xb9, 0xc5, 0x89}};
```

File: src/coreclr/gc/gcpriv.h [hunk 3]
```diff
@@ -92,10 +92,10 @@ inline void FATAL_GC_ERROR()
 #define DOUBLY_LINKED_FL
 #endif //HOST_64BIT
 
-#ifndef FEATURE_REDHAWK
+#ifndef FEATURE_NativeAOT
```
Comment (jkotas) [inline src/coreclr/gc/gcpriv.h:95]: This should maintain the casing, ie this should be `FEATURE_NATIVEAOT`

File: src/coreclr/nativeaot/Runtime/HandleTableHelpers.cpp [hunk 4]
```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 //
-// Helper functions that are p/invoked from redhawkm in order to expose handle table functionality to managed
+// Helper functions that are p/invoked from NativeAOTm in order to expose handle table functionality to managed
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/HandleTableHelpers.cpp:5]: `NativeAOTm` is not a thing. This is example, of place that needs manual tweaks.

**Comment by jkotas** [comment]
Check my suggestion at https://github.com/dotnet/runtime/issues/69115#issuecomment-1123871555 about what I believe is the most productive way to do this.

**Pull request closed without merging**